aws/rakefiles/deploy.rake

```diff
@@ -23,14 +23,19 @@ task :wait_for_gpii_ready => :configure_kubectl do
   # Test preferences server and cloud based flow manager
   preferences_test = {
     :url => "preferences.#{ENV["TF_VAR_cluster_name"]}/preferences/carla",
-    :curl_options => ""
+    :curl_options => "",
+    :cert_resource_name => "preferences-cert"
   }
   flowmanager_test = {
     :url => "flowmanager.#{ENV["TF_VAR_cluster_name"]}/access_token",
-    :curl_options => " -H 'Content-Type: application/x-www-form-urlencoded' -X POST -d 'username=carla&password=dummy&client_id=pilot-computer&client_secret=pilot-computer-secret&grant_type=password' -X POST "
+    :curl_options => " -H 'Content-Type: application/x-www-form-urlencoded' -X POST -d 'username=carla&password=dummy&client_id=pilot-computer&client_secret=pilot-computer-secret&grant_type=password' -X POST ",
+    :cert_resource_name => "flowmanager-cert"
   }
 
   [preferences_test, flowmanager_test].each do |test|
+    # First we wait for certificate object to be ready
+    puts "Waiting for certificates from Let's Encrypt"
+    wait_for("kubectl get certificate #{test[:cert_resource_name]} --namespace gpii --output json | jq -e '.status.conditions[] | select(.type == \"Ready\" and .status == \"True\")'")
```

Comment:
By default, `wait_for` waits 15 minute (and retries every 20 seconds). Just checking if these are reasonable defaults here.

Reply:
I think this is reasonable. If the cert is not ready in 15 minutes, something is seriously wrong. And 20s retries sounds like a decent compromise between trying too often and waiting too long.